docs/source/codependence/information_theory_metrics.rst

```diff
@@ -0,0 +1,236 @@
+
+.. math::
+
+    I[X, Y, Z] = I[X, Y] + I[(X, Y), Z]
+
+where :math:`(X, Y)` is a joint distribution of :math:`X` and :math:`Y` .
+
+It can also be normalized using a known upper boundary:
+
+.. math::
+
+    I[X, Y] \le min\{H[X] + H[Y]\}
+
+Implementation
+##############
+
+.. py:currentmodule:: mlfinlab.codependence.information
+
+.. autofunction:: get_mutual_info
+
+
+Variation of Information
+========================
+
+According to Lopez de Prado: "**Variation of Information** can be interpreted as the uncertainty we expect in one variable
+if we are told the value of another". The variation of information is a true metric and satisfies the axioms from the introduction.
+
+.. math::
+   \begin{align*}
+     VI[X,Y]=& H[X|Y] + H[Y|X]\\
+           =& H[X] + H[Y]-2I[X,Y]\\
+           =& 2H[X,Y]-H[X]-H[Y]\\
+   \end{align*}
+
+The upper bound of Variation of information is not firm as it depends on the sizes of the population which is problematic
+when comparing variations of information across different population sizes, as described in
+`Cornell lecture slides, p.21 <https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3512994>`_
+
+Implementation
+##############
+
+.. autofunction::  variation_of_information_score
+
+
+Discretization
+==============
+
+Both mutual information and variation of information are using random variables that are discrete. To use these tools for
+continuous random variables the discretization approach can be used.
+
+For the continuous case, we can quantize the values to estimate :math:`H[X]`. Following the `Cornell lecture slides, p.26 <https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3512994>`_ :
+
+.. math::
+    \begin{align*}
+    H[X] =& \int_{\infty}^{\infty}f_{X}[x]log[f_{X}[x]]dx\\
+    \:    \approx& -\sum\limits_{i=1}^{B_{X}}f_{X}[x_{i}]log[f_{X}[x_{i}]]\Delta_{x}\\
+    \end{align*}
+
+where the observed values :math:`\{x\}` are divided into :math:`B_{X}` bins of equal size :math:`\Delta_{X}`,
+:math:`\Delta_{X} = \frac{max\{x\} - min\{x\}}{B_{X}}` , and :math:`f_{X}[x_{i}]` is the frequency of observations
+within the i-th bin.
+
+So, the discretized estimator of entropy is:
+
+.. math::
+    \hat{H}[X]=-\sum\limits_{i=1}^{B_{X}}\frac{N_{i}}{N}log[\frac{N_{i}}{N}]log[\Delta_{X}]
+
+where :math:`N_{i}` is the number of observations within the i-th bin, :math:`N = \sum_{i=1}^{B_{X}}N_{i}` .
+
+From the above equations, the size of the bins should be chosen. The results of the entropy estimation will depend on the
+binning. The works by `Hacine-Gharbi et al. (2012) <https://www.researchgate.net/publication/257014935>`_  and
+`Hacine-Gharbi and Ravier (2018) <https://www.researchgate.net/publication/320887281>`_  present optimal binning
+for marginal and joint entropy.
+
+This optimal binning method is used in the mutual information and variation of information functions.
+
+Implementation
+##############
+
+.. autofunction:: get_optimal_number_of_bins
+
+
+Examples
```

Comment:
Can you change this to "Example Code", just to be more clear for the users

Reply:
I was just following the style that we use in the PortOpt and other modules. 

I think I'll leave it like this for now and then make a PR to change the names of Code examples globally in MlFinLab. 

Thank you for noticing!